pkg/upload/dsym.go

```diff
@@ -36,82 +38,105 @@ func ProcessDsym(
+			log.Info("Defaulting to '" + projectRoot + "' as the project root")
+
+			// Get build settings and dsymPath
+
+			// If scheme is set explicitly, check if it exists
+			if scheme != "" {
+				_, err := ios.IsSchemeInPath(path, scheme)
+				if err != nil {
+					log.Warn(err.Error())
+				}
 
-			// Only when the dsym path is not set, try and work out the scheme
-			if dsymPath == "" && len(uploadInfo.DsymPaths) == 0 {
-				// If the scheme is not set explicitly, try to find it
+			} else {
+				// Otherwise, try to find it
+				var err error
 				scheme, err = ios.GetDefaultScheme(path)
 				if err != nil {
-					return err
+					log.Warn(err.Error())
 				}
-				log.Info("Using scheme: " + scheme)
+
 			}
-		}
 
-		// If the dsymPath is not fed in via <path>
-		if dsymPath == "" && len(uploadInfo.DsymPaths) == 0 {
+			var err error
 			buildSettings, err = ios.GetXcodeBuildSettings(path, scheme)
 			if err != nil {
 				return err
 			}
 
-			// Build the dsymPath from build settings
-			// Which is built up to look like: /Users/Path/To/Config/Build/Dir/MyApp.app.dSYM/Contents/Resources/DWARF
-			dsymPath = filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.DsymName, "Contents", "Resources", "DWARF")
-
-			// Check if dsymPath exists before proceeding
-			if utils.Path(dsymPath).Validate() != nil {
-				// TODO: This will be downgraded to a warning with --ignore-missing-dwarf in near future
-				log.Error("Could not find dSYM with scheme '"+scheme+"' in expected location: "+utils.DisplayBlankIfEmpty(dsymPath)+"\n\n"+
-					"Check that the scheme correlates to the above dSYM location, try re-building your project or specify the dSYM path using --dsym-path", 1)
-			} else {
-				log.Info("Using dSYM path: " + dsymPath)
+			if dsymPath == "" {
+				// Build the dsymPath from build settings
+				// Which is built up to look like: /Users/Path/To/Config/Build/Dir/MyApp.app.dSYM
+				dsymPath = filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.DsymName)
+
+				// Check if dsymPath exists before proceeding
+				if utils.Path(dsymPath).Validate() != nil {
+					// TODO: This will be toggled between Error and Warn with --ignore-missing-dwarf in near future
+					log.Error("Could not find dSYM with scheme '" + scheme + "' in expected location: " + utils.DisplayBlankIfEmpty(dsymPath) + "\n\n" +
+						"Check that the scheme correlates to the above dSYM location, try re-building your project or specify the dSYM path using --dsym-path\n", 1)
+				} else {
+					log.Info("Using dSYM path: " + dsymPath)
+				}
 			}
 
+		} else if dsymPath == "" {
+			log.Info("No Xcode project, workspace or package in '" + path + "'")
+			dsymPath = path
 		}
 
-		dsyms, err = ios.GetDsymsForUpload(uploadInfo.DsymPaths)
-		if err != nil {
-			return err
+		if dsymPath != "" {
+			var tempDir string
+			dwarfInfo, tempDir, _ = ios.FindDsymsInPath(dsymPath)
+			if len(dwarfInfo) > 0 && projectRoot == "" {
```

Comment:
This might read a bit better as an else-if on L121 (after the zero length check)